primer3/thermoanalysis.pyx

```diff
@@ -949,3 +938,1009 @@ cdef class ThermoAnalysis:
+            ):
+                sqtemp_b = <bytes> retval[0].best_pairs.pairs[i].compl_end_struct
+                output_dict[f'PRIMER_PAIR_{i}_COMPL_END_STUCT'] = sqtemp_b.decode('utf8')
+
+            # Print product size
+            product_size = retval[0].best_pairs.pairs[i].product_size
+            if sequence_args_data[0].overhang_left != NULL:
+                product_size += strlen(sequence_args_data[0].overhang_left)
+            if sequence_args_data[0].overhang_right != NULL:
+                product_size += strlen(sequence_args_data[0].overhang_right)
+            output_dict[f'PRIMER_PAIR_{i}_PRODUCT_SIZE'] = product_size
+
+
+            # Print the product Tm if a Tm range is defined
+            if (
+                (global_settings_data[0].product_max_tm != PR_DEFAULT_PRODUCT_MAX_TM) or
+                (global_settings_data[0].product_min_tm != PR_DEFAULT_PRODUCT_MIN_TM)
+            ):
+                output_dict[f'PRIMER_PAIR_{i}_PRODUCT_TM'] = \
+                    retval[0].best_pairs.pairs[i].product_tm
+                output_dict[f'PRIMER_PAIR_{i}_PRODUCT_TM_OLIGO_TM_DIFF'] = \
+                    retval[0].best_pairs.pairs[i].product_tm_oligo_tm_diff
+                output_dict[f'PRIMER_PAIR_{i}_T_OPT_A'] = retval[0].best_pairs.pairs[i].t_opt_a
+            else:
+                output_dict[f'PRIMER_PAIR_{i}_PRODUCT_TM'] = \
+                retval[0].best_pairs.pairs[i].product_tm
+
+            # Print the primer pair template mispriming
+            if (
+                (global_settings_data[0].thermodynamic_template_alignment == 0) and
+                (retval[0].best_pairs.pairs[i].template_mispriming != ALIGN_SCORE_UNDEF)
+            ):
+                output_dict[f'PRIMER_PAIR_{i}_TEMPLATE_MISPRIMING'] = \
+                    retval[0].best_pairs.pairs[i].template_mispriming
+            # Print the primer pair template mispriming. Thermodynamic approach.
+            if (
+                    (global_settings_data[0].thermodynamic_template_alignment == 1) and
+                    (retval[0].best_pairs.pairs[i].template_mispriming != ALIGN_SCORE_UNDEF)
+            ):
+                output_dict[f'PRIMER_PAIR_{i}_TEMPLATE_MISPRIMING_TH'] = \
+                    retval[0].best_pairs.pairs[i].template_mispriming
+
+            # Print primer secondary structures*/
+            if (
+                 (global_settings_data[0].show_secondary_structure_alignment == 1) and
+                (retval[0].best_pairs.pairs[i].template_mispriming_struct != NULL)
+            ):
+                sqtemp_b = <bytes> retval[0].best_pairs.pairs[i].template_mispriming_struct
+                output_dict[f'PRIMER_PAIR_{i}_TEMPLATE_MISPRIMING_STUCT'] = \
+                    sqtemp_b.decode('utf8')
+        # End of print parameters of primer pairs
+    # End of the for big loop printing all data
+    return output_dict
+
+
+def design_clean_up():
```

Comment:
Semantic consideration -- now that we have a unified cython file, what's the generic terminology for common functions like this? I think we used to delineate between `design` and `analysis` (`thermoanalysis`?). 

If I'm following this PR correctly, there are now unified `primer3` library global initialization and cleanup functions, of which this is one. So maybe they should be grouped together and titled w/ some common prefix (e.g., `p3_lib_init` and `p3_lib_deinit` or similar)

Reply:
This suggestion makes sense to me.